src/Stratis.Features.FederatedPeg/TargetChain/CrossChainTransferStore.cs

```diff
@@ -1256,6 +1310,37 @@ public bool ValidateTransaction(Transaction transaction, bool checkSignature = f
             }
         }
 
+        /// <summary>
+        /// Determines if a mempool error would NOT be recoverable by waiting or rebuilding the transaction.
+        /// </summary>
+        /// <param name="mempoolError">The error to evailuate.</param>
+        /// <returns><c>true</c> if its irrecoverble or <c>false</c> otherwise.</returns>
+        public static bool IsMempoolErrorRecoverable(MempoolError mempoolError)
+        {
+            switch (mempoolError.RejectCode)
+            {
+                // Can recover from inputs already spent.
+                case MempoolErrors.RejectDuplicate:
+                    return true;
+
+                // Duplicates should not make the transfer fail.
+                case MempoolErrors.RejectAlreadyKnown:
+                    return true;
+
+                default:
+                    switch (mempoolError.ConsensusError?.Code)
```

Comment:
A nested switch statement here is highly unreadable. And `Code` can never be null - so this could just be written as
`mempoolError.ConsensusError == null`.

Reply:
`Code` can be `null` if the default constructor of `MempoolError` is used. I am leaving this as a `switch` for now as we likely will think of more cases to add in the future.

Reply:
We don't have more cases to add right now and might never have, so why plan prematurely for that and sacrifice readability?

Reply:
Done.